integration/run_helpers.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/columns"
+	columns_json "github.com/inspektor-gadget/inspektor-gadget/pkg/columns/formatter/json"
+	runtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/run/types"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+func MarshalJson(t *testing.T, m map[string]interface{}) string {
+	b, err := json.Marshal(m)
+	require.NoError(t, err, "marshalling json")
+	return string(b)
+}
+
+func SetEventTimestamp(jsonObj map[string]interface{}, timestamp eventtypes.Time) {
+	jsonObj["timestamp"] = timestamp.String()
+}
+
+func SetEventMountNsID(jsonObj map[string]interface{}, mountNsID uint64) {
+	jsonObj["mntns"] = mountNsID
+}
+
+func SetEventRuntimeName(jsonObj map[string]interface{}, runtimeName eventtypes.RuntimeName) {
+	runtimeMetadata := jsonObj["runtime"].(map[string]interface{})
+	if runtimeMetadata != nil {
+		runtimeMetadata["runtimeName"] = runtimeName
+	}
+}
+
+func SetEventRuntimeContainerID(jsonObj map[string]interface{}, s string) {
+	runtimeMetadata := jsonObj["runtime"].(map[string]interface{})
+	if runtimeMetadata != nil {
+		runtimeMetadata["containerId"] = s
+	}
+}
+
+func SetEventRuntimeContainerName(jsonObj map[string]interface{}, s string) {
+	runtimeMetadata := jsonObj["runtime"].(map[string]interface{})
+	if runtimeMetadata != nil {
+		runtimeMetadata["containerName"] = s
+	}
+}
+
+func SetEventK8sNode(jsonObj map[string]interface{}, s string) {
+	jsonObj["node"] = s
+}
+
+func SetEventK8sNamespace(jsonObj map[string]interface{}, s string) {
+	jsonObj["namespace"] = s
+}
+
+func SetEventK8sPod(jsonObj map[string]interface{}, s string) {
+	jsonObj["pod"] = s
+}
+
+func SetEventK8sContainer(jsonObj map[string]interface{}, s string) {
+	jsonObj["container"] = s
+}
+
+func SetEventK8sHostNetwork(jsonObj map[string]interface{}, b bool) {
+	jsonObj["hostnetwork"] = b
+}
+
+func RunEventToJsonObj(t *testing.T, ev *runtypes.Event) map[string]interface{} {
```

Comment:
I have some doubts about this function's name because it only takes into consideration "hardcoded" columns and not the ones each gadget uses. In fact it just ignores (and removes) the `raw_data` and `data` fields. 
Can't we use the standard json marshaller for that?

Reply:
Unfortunately we can't use the standard json marshaller since we need to use the column names instead of the tagged json names